interactive_marker_tutorials/scripts/basic_controls.py

```diff
@@ -1,109 +1,147 @@
-        rospy.loginfo( s + ": menu item " + str(feedback.menu_entry_id) + " clicked" + mp + "." )
+        node.get_logger().info(
+            '{log_prefix}: menu item {feedback.menu_entry_id} clicked at {log_mouse}'
+            .format_map(locals())
+        )
     elif feedback.event_type == InteractiveMarkerFeedback.POSE_UPDATE:
-        rospy.loginfo( s + ": pose changed")
-# TODO
-#          << "\nposition = "
-#          << feedback.pose.position.x
-#          << ", " << feedback.pose.position.y
-#          << ", " << feedback.pose.position.z
-#          << "\norientation = "
-#          << feedback.pose.orientation.w
-#          << ", " << feedback.pose.orientation.x
-#          << ", " << feedback.pose.orientation.y
-#          << ", " << feedback.pose.orientation.z
-#          << "\nframe: " << feedback.header.frame_id
-#          << " time: " << feedback.header.stamp.sec << "sec, "
-#          << feedback.header.stamp.nsec << " nsec" )
+        node.get_logger().info(
+            '{log_prefix}: pose changed\n'
+            'position: '
+            '{feedback.pose.position.x}, {feedback.pose.position.y}, {feedback.pose.position.z}\n'
+            'orientation: '
+            '{feedback.pose.orientation.w}, {feedback.pose.orientation.x}, '
+            '{feedback.pose.orientation.y}, {feedback.pose.orientation.z}\n'
+            'frame: {feedback.header.frame_id} '
+            'time: {feedback.header.stamp.sec} sec, '
+            '{feedback.header.stamp.nanosec} nsec'
+            .format_map(locals())
+        )
     elif feedback.event_type == InteractiveMarkerFeedback.MOUSE_DOWN:
-        rospy.loginfo( s + ": mouse down" + mp + "." )
+        node.get_logger().info('{log_prefix}: mouse down at {log_mouse}'.format_map(locals()))
     elif feedback.event_type == InteractiveMarkerFeedback.MOUSE_UP:
-        rospy.loginfo( s + ": mouse up" + mp + "." )
-    server.applyChanges()
+        node.get_logger().info('{log_prefix}: mouse up at {log_mouse}'.format_map(locals()))
 
-def alignMarker( feedback ):
+
+def alignMarker(feedback):
     pose = feedback.pose
 
-    pose.position.x = round(pose.position.x-0.5)+0.5
-    pose.position.y = round(pose.position.y-0.5)+0.5
+    pose.position.x = round(pose.position.x - 0.5) + 0.5
+    pose.position.y = round(pose.position.y - 0.5) + 0.5
 
-    rospy.loginfo( feedback.marker_name + ": aligning position = " + str(feedback.pose.position.x) + "," + str(feedback.pose.position.y) + "," + str(feedback.pose.position.z) + " to " +
-                                                                     str(pose.position.x) + "," + str(pose.position.y) + "," + str(pose.position.z) )
+    node.get_logger().info(
+        '{feedback.marker_name}: aligning position = {feedback.pose.position.x}, '
+        '{feedback.pose.position.y}, {feedback.pose.position.z} to '
+        '{pose.position.x}, {pose.position.y}, {pose.position.z}'
+        .format_map(locals())
```

Comment:
If this is Eloquent only then we can use Python 3.6 features (https://www.ros.org/reps/rep-2000.html#eloquent-elusor-november-2019-november-2020) and therefore you could use the f-string feature: https://realpython.com/python-f-strings/#f-strings-a-new-and-improved-way-to-format-strings-in-python

Reply:
:cool: I'll try that.

Reply:
fc450bb